crates/web5/src/test_vectors.rs

```diff
@@ -0,0 +1,210 @@
+                    "Document metadata does not match."
+                );
+            }
+        }
+    }
+
+    mod did_dht {
+        use super::*;
+        use crate::dids::{
+            methods::did_dht::DidDht,
+            resolution::resolution_metadata::{ResolutionMetadata, ResolutionMetadataError},
+        };
+
+        #[derive(Debug, PartialEq, serde::Deserialize)]
+        struct VectorInput {
+            #[serde(rename = "didUri")]
+            did_uri: String,
+        }
+
+        #[derive(Debug, PartialEq, serde::Deserialize)]
+        struct VectorOutput {
+            #[serde(rename = "didResolutionMetadata")]
+            did_resolution_metadata: ResolutionMetadata,
+        }
+
+        #[test]
+        fn resolve() {
+            let path = "did_dht/resolve.json";
+            let vectors: TestVectorFile<VectorInput, VectorOutput> =
+                TestVectorFile::load_from_path(path);
+
+            for vector in vectors.vectors {
+                let vector_input = vector.input;
+                let vector_output = &vector.output;
+
+                // As a replay attack protection protocol, if the same DID is doing a GET request within 5 minutes of each other, instead of a 404 it will start returning a 429.
+                // to get around this for our test we just create a new DID that is not published to get a fresh 404 for this error code
+                if let Some(ResolutionMetadataError::NotFound) =
+                    vector_output.did_resolution_metadata.error
+                {
+                    // TODO: According to the did dht spec a 404 should be returned when trying to resolve a DID that does not exists. Currently it incorrectly returns a 429 even on the first call.
+                    // Uncomment this code block when resolved - https://github.com/TBD54566975/web5-rs/issues/286
+                    continue;
+
+                    // let private_jwk = Ed25519Generator::generate();
+                    // let identity_key = ed25519::to_public_jwk(&private_jwk);
+                    // let did_dht =
+                    //     DidDht::from_identity_key(identity_key.clone()).expect("Should create did:dht");
+                    //
+                    // vector_input = VectorInput{
+                    //     did_uri: did_dht.did.uri,
+                    // };
+                }
+
+                let resolution_result = DidDht::resolve(&vector_input.did_uri);
+
+                let metadata_error = resolution_result.resolution_metadata.error.as_ref();
+                let expected_error = vector_output.did_resolution_metadata.error.as_ref();
+
+                assert_eq!(
+                    metadata_error, expected_error,
+                    "Document resolution metadata does not match. Expected '{:?}' but got '{:?}'.",
+                    expected_error, metadata_error
+                );
+            }
+        }
+    }
+
+    mod presentation_definition {
```

Comment:
```suggestion
    mod presentation_exchange {
```



needs 
<img width="463" alt="image" src="https://github.com/user-attachments/assets/d47df6a3-3881-4ecf-9d40-817f1e6c4f49">

Reply:
nice catch, fixed